### PR TITLE
[9137] Removing C++17 warnings from sources

### PR DESCRIPTION
**Pull request by MiguelBarro**
This pull request addresses the issue #1275.

C++ 17 deprecated classes were removed and replaced with compliant code:
- std::iterator and std::unary_function as base classes.
- std::wstring_convert as string conversion utility.

Note that Fast-DDS is not yet fully complaint because the boost sources introduced for shared memory transport sake are not. In order to avoid the use of boost sources:
- cmake variable `SHM_TRANSPORT_DEFAULT=OFF`
- set only /std:c++17 for the Fast-DDS sources only by adding to the CMakeLists.txt:
```cmake
 target_compile_options(${PROJECT_NAME} PRIVATE /std:c++17)
```

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2511/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2324/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2518/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2520/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2333/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2527/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2528/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2339/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2535/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2533/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2344/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2540/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2606/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2422/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2616/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2667/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/72/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2473/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2677/)

**Review by EduPonz — CHANGES_REQUESTED** [review]
Please uncrustify

**Comment by MiguelCompany** [comment]
@imontesino  This has to be rebased

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2806/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/151/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2576/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2818/)

File: include/fastrtps/utils/fixed_size_bitmap.hpp [hunk 1]
```diff
@@ -20,6 +20,7 @@
 #ifndef FASTRTPS_UTILS_FIXED_SIZE_BITMAP_HPP_
 #define FASTRTPS_UTILS_FIXED_SIZE_BITMAP_HPP_
 
+#include <cstdint>
```
Comment (MiguelCompany) [inline include/fastrtps/utils/fixed_size_bitmap.hpp:23]: This line is duplicated. Please remove it.

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2816/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/158/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2586/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2828/)

**Review by MiguelCompany — APPROVED** [review]

**Comment by MiguelCompany** [comment]
Not waiting for tests, since there have only been linter changes after the last ones, which were OK.

**Pull request merged**